### DISCUSSION TITLE
useContractRead can support timer?

### DISCUSSION THREAD

**Discussion by lukema95** [wevm/wagmi#1109] [Q&A]
I want to automatically get data once every 1s. What should I do?
```
const {data} = useContractRead({
    addressOrName: contractConfig.address,
    contractInterface: contractABI,
    functionName: 'totalSupply',
    onSuccess(data) {
      console.log('Get totalSupply success', data)
    },
    chainId: contractConfig.chainID,
    onError(error) {
      console.log('Get totalSupply error', error)
    },
  })
```

**Comment by tmm** [comment]
Check out the [`watch`](https://wagmi.sh/docs/hooks/useContractRead#watch-optional) configuration parameter. It watches for new blocks (when there is new data) instead of polling every second (might not have new data).

**Reply by johnpaulcas** [reply]
hi @tmm Im using add but my issue is that I always getting 429 issue on my provider, any solution for this?

**Reply by tmm** [reply]
@johnpaulcas do you have a [minimal reproducible example](
https://stackoverflow.com/help/minimal-reproducible-example) to check out?

**Reply by johnpaulcas** [reply]
> @johnpaulcas do you have a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to check out?

whats the best practice using watch? currently im using watch on mapping of item, where each mapping watch blockchain status thats why Im getting this issue, any way to make the interval of watch to be like every few seconds before next request?

thanks
